Robinhood/Robinhood.py

```diff
@@ -829,52 +967,68 @@ def place_order(self,
                 instrument (dict): the RH URL and symbol in dict for the instrument to be traded
                 quantity (int): quantity of stocks in order
                 bid_price (float): price for order
-                transaction (:enum:`Transaction`): BUY or SELL enum
-                trigger (:enum:`Trigger`): IMMEDIATE or STOP enum
+                stop_price (float): price at which order is placed
+                transaction (:enum:`Transaction`): BUY or SELL
+                trigger (:enum:`Trigger`): IMMEDIATE or STOP
                 order (:enum:`Order`): MARKET or LIMIT
-                time_in_force (:enum:`TIME_IN_FORCE`): GFD or GTC (day or until cancelled)
+                time_in_force (:enum:`TimeForce`): GFD or GTC (day or until cancelled)
 
             Returns:
                 (:obj:`requests.request`): result from `orders` put command
         """
+        # Insure the order instructions are valid
+        transaction = self.Transaction(transaction)
+        trigger = self.Trigger(trigger)
+        order = self.Order(order)
+        time_in_force = self.TimeForce(time_in_force)
+
+        if order == self.Order.LIMIT and bid_price is None:
+            raise ValueError("Order.LIMIT without bid_price")
 
-        if isinstance(transaction, str):
-            transaction = Transaction(transaction)
+        if bid_price is not None and not order == self.Order.LIMIT:
+            raise ValueError("bid_price without Order.LIMIT")
 
-        if not bid_price:
-            bid_price = self.quote_data(instrument['symbol'])['bid_price']
+        # If Order.MARKET then we still need to pass RH a price, RH places market orders
+        # automatically as limit orders with a 5% buffer
+        if not bid_price and order == self.Order.MARKET:
+            bid_price = bid_price(instrument['symbol'])
 
         payload = {
             'account': self.get_account()['url'],
             'instrument': unquote(instrument['url']),
+            'quantity': int(quantity),
             'price': float(bid_price),
-            'quantity': quantity,
-            'side': transaction.name.lower(),
+
+            'side': transaction.value,
             'symbol': instrument['symbol'],
-            'time_in_force': time_in_force.lower(),
-            'trigger': trigger,
-            'type': order.lower()
+            'time_in_force': time_in_force.value,
+            'trigger': trigger.value,
+            'type': order.value
         }
 
-        #data = 'account=%s&instrument=%s&price=%f&quantity=%d&side=%s&symbol=%s#&time_in_force=gfd&trigger=immediate&type=market' % (
-        #    self.get_account()['url'],
-        #    urllib.parse.unquote(instrument['url']),
-        #    float(bid_price),
-        #    quantity,
-        #    transaction,
-        #    instrument['symbol']
-        #)
-
-        res = self.session.post(self.endpoints['orders'], data=payload)
-        res.raise_for_status()
+        if trigger == self.Trigger.STOP:
```

Comment:
another if/then tree.  Can the complexity be unloaded up into the `enum` and allowed to raise if bad trigger instead?